mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp

```diff
@@ -349,14 +361,62 @@ class AffineForLowering : public OpRewritePattern<AffineForOp> {
     Value lowerBound = lowerAffineLowerBound(op, rewriter);
     Value upperBound = lowerAffineUpperBound(op, rewriter);
     Value step = rewriter.create<ConstantIndexOp>(loc, op.getStep());
-    auto f = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step);
+    SmallVector<Value, 4> iterArgs(op.getIterOperands().begin(),
+                                   op.getIterOperands().end());
+    scf::ForOp f;
+    if (iterArgs.size() > 0) {
+      f = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step,
+                                      iterArgs);
+    } else {
+      f = rewriter.create<scf::ForOp>(loc, lowerBound, upperBound, step);
+    }
     rewriter.eraseBlock(f.getBody());
     rewriter.inlineRegionBefore(op.region(), f.region(), f.region().end());
-    rewriter.eraseOp(op);
+    rewriter.replaceOp(op, f.results());
     return success();
   }
 };
 
+/// This function returns the value of identity value associated with a
```

Comment:
Drop redundant "This function ..". Just "Returns the value ...".

Reply:
Done.